.github/workflows/scripts/prs/detectTargetBranch.js

```diff
@@ -0,0 +1,105 @@
+// @ts-check
+const vBranchRegex = /^v\d{1,3}\.x$/;
+const transferLabels = ['cherry-pick'];
+
+/**
+ * @param {Object} params
+ * @param {import("@actions/core")} params.core
+ * @param {ReturnType<import("@actions/github").getOctokit>} params.github
+ * @param {import("@actions/github").context} params.context
+ */
+module.exports = async ({ core, context, github }) => {
+  try {
+    const owner = context.repo.owner;
+    const repo = context.repo.repo;
+    const pullNumber = context.issue.number;
+
+    const pr = await github.rest.pull_request.get({
+      owner,
+      repo,
+      pull_number: pullNumber,
+    });
+
+    core.info(`>>> PR fetched: ${pr.id}`);
+
+    const targetLabels = pr.labels
+      .map((label) => label.name)
+      .filter((label) => vBranchRegex.test(label));
+
+    if (targetLabels.length === 0) {
+      // there was no target branch present
+      core.info('>>> No target branch label found');
+
+      if (vBranchRegex.test(pr.head_ref)) {
```

Comment:
Am I missing something, but which case is this handling?
We cherry-pick from `master` to v*.x (i.e. v7.x) or  from `v*.x` to `v*-1.x`, but seems to set target branch to `master` when we would cherry-pick from `v7.x` 🤔

Reply:
yes, this is the case where you cherry pick from `vx.x` to `master` ... its basically option 3 from the PR description